src/lib/y2storage/boot_requirements_strategies/base.rb

```diff
@@ -22,61 +22,61 @@
 # find current contact information at www.suse.com.
 
 require "yast"
-require "storage/patches"
 require "y2storage/disk_size"
 require "y2storage/filesystems/type"
 require "y2storage/planned"
+require "y2storage/boot_requirements_strategies/analyzer"
+require "y2storage/exceptions"
 
 module Y2Storage
   module BootRequirementsStrategies
-    class Error < RuntimeError
+    class Error < Y2Storage::Error
     end
 
     # Base class for the strategies used to calculate the boot partitioning
     # requirements
     class Base
       include Yast::Logger
+      extend Forwardable
 
-      def initialize(settings, devicegraph)
-        @settings = settings
+      def_delegators :@analyzer,
+        :boot_disk, :root_in_lvm?, :encrypted_root?, :btrfs_root?, :boot_ptable_type?, :free_mountpoint?
+
+      # Constructor
+      #
+      # @see [BootRequirementsChecker#devicegraph]
+      # @see [BootRequirementsChecker#planned_devices]
+      # @see [BootRequirementsChecker#boot_disk_name]
+      def initialize(devicegraph, planned_devices, boot_disk_name)
         @devicegraph = devicegraph
-        @root_disk = devicegraph.disks.detect { |d| d.name == settings.root_device }
+        @analyzer = Analyzer.new(devicegraph, planned_devices, boot_disk_name)
       end
 
       def needed_partitions(target)
-        boot_partition_needed? ? [boot_partition(target)] : []
+        return [] unless boot_partition_needed? && boot_partition_missing?
+        [boot_partition(target)]
       end
 
     protected
 
-      attr_reader :settings
       attr_reader :devicegraph
-      attr_reader :root_disk
+      attr_reader :analyzer
 
       def boot_partition_needed?
         false
       end
 
+      def boot_partition_missing?
+        free_mountpoint?("/boot")
+      end
+
       def boot_partition(target)
         vol = Planned::Partition.new("/boot", Filesystems::Type::EXT4)
-        vol.disk = settings.root_device
+        vol.disk = boot_disk.name
         vol.min_size = target == :min ? DiskSize.MiB(100) : DiskSize.MiB(200)
         vol.max_size = DiskSize.MiB(500)
```

Comment:
And about the usage of constants, this is again re-reviewing quite old code just because the line before was modified. This PR is already big and complex enough. :wink: